cpp/src/strings/convert/int_cast.cu

```diff
@@ -0,0 +1,276 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/detail/null_mask.hpp>
+#include <cudf/detail/nvtx/ranges.hpp>
+#include <cudf/strings/convert/int_cast.hpp>
+#include <cudf/strings/detail/strings_children.cuh>
+#include <cudf/strings/string_view.cuh>
+#include <cudf/strings/strings_column_view.hpp>
+#include <cudf/utilities/default_stream.hpp>
+#include <cudf/utilities/memory_resource.hpp>
+#include <cudf/utilities/traits.hpp>
+#include <cudf/utilities/type_dispatcher.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/exec_policy.hpp>
+
+#include <cuda/std/functional>
+#include <thrust/for_each.h>
+#include <thrust/iterator/counting_iterator.h>
+#include <thrust/transform_reduce.h>
+
+namespace cudf {
+namespace strings {
+namespace detail {
+
+namespace {
+
+struct dispatch_set_int_fn {
+  template <typename T>
+    requires(cudf::is_integral_not_bool<T>())
+  __device__ void operator()(mutable_column_device_view& d_results,
+                             size_type idx,
+                             uint64_t value) const
+  {
+    d_results.element<T>(idx) = static_cast<T>(value);
+  }
+  template <typename T>
+    requires(not cudf::is_integral_not_bool<T>())
+  __device__ void operator()(mutable_column_device_view&, size_type, uint64_t) const
+  {
+    CUDF_UNREACHABLE("invalid type");
+  }
+};
+
+struct cast_to_integer_fn {
+  column_device_view const d_strings;
+  mutable_column_device_view d_results;
+  endian swap;
+  size_type output_type_size;
+
+  __device__ void operator()(size_type idx)
+  {
+    if (d_strings.is_null(idx)) { return; }
+    auto const d_str = d_strings.element<string_view>(idx);
+    auto const size  = std::min(d_str.size_bytes(), output_type_size);
+
+    auto value = uint64_t{0};
+    auto data  = reinterpret_cast<u_char const*>(d_str.data());
+    if (swap == endian::LITTLE) {
+      for (size_type i = 0; i < size; i++) {
```

Comment:
There are intrinsics that do this endian swap for 32 bit sizes and probably other sizes too. Should we use those?

I think the cost of byte loads vs. vectorized word loads could be significant.

Reply:
The intrinsic `byteperm` only works here if `size==32` exactly. I could add a specific if-statement but I'm not sure the divergence is worth it.

Reply:
Okay. I thought there might be similar intrinsic funcitions for other sizes. I'd like to know if this makes a difference in performance but it's not required for me to approve.